oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/introspection/SpringOpaqueTokenIntrospector.java

```diff
@@ -58,16 +57,16 @@
  */
 public class SpringOpaqueTokenIntrospector implements OpaqueTokenIntrospector {
 
-	private final Log logger = LogFactory.getLog(getClass());
+	private static final String AUTHORITY_PREFIX = "SCOPE_";
```

Comment:
You use 2nd-time AUTHORITY_PREFIX  as constant. The first one was in NimbusReactiveOpaqueTokenIntrospector.java. Is it better to extract constant in one place? What are you thinking about it?

Reply:
I was thinking this, but it appears that this constant is used in many more places than just here. Since this was already in the file, I just moved the line it was on.

I was tempted to do this but I didn't want to make this PR too large and increase the scope of it too much. I was considering whether or not to draft a separate PR once this fix has gone through to get all the declarations for "ROLE_" and "SCOPE_" and put them in one place.

Reply:
Oh, I see. You are absolutely right about not blowing this PR. Good idea about extracting "ROLE_" and "SCOPE_". For me, it looks like an independent PR.